Il2CppInspector.Common/Reflection/TypeInfo.cs

```diff
@@ -763,7 +814,7 @@ private class TypeArgumentsComparer : EqualityComparer<TypeInfo[]>
         // and returns a TypeInfo object representing the resulting constructed type.
         // See: https://docs.microsoft.com/en-us/dotnet/api/system.type.makegenerictype?view=netframework-4.8
         public TypeInfo MakeGenericType(params TypeInfo[] typeArguments) {
-            if(typeArguments.Length != genericArguments.Length) {
+            if (typeArguments.Length != genericArguments.Length) {
                 throw new ArgumentException("The number of generic arguments provided does not match the generic type definition.");
             }
 
```

Comment:
This commit breaks most of the tests from my test suite of games. The error is more or less the same for every binary, outputting a `new` modifier on certain method or property signatures where none should exist (I believe). I haven't gone over the code so perhaps you could look into it. Let me know if you need test samples.

```
 Message: 
      Expected and actual are both <System.String[161937]>
      Values differ at index [151508]
      Expected string length 137 but was 141. Strings differ at index 7.
      Expected: "public UnityEngine.Object value { get => default; set {} } //..."
      But was:  "public new UnityEngine.Object value { get => default; set {} ..."
      ------------------^
```

Reply:
Commit 1ac6395c6ba4cb271cf9bc97873330d70f8f49ad fixes these issues and the testcases.

Reply:
It doesn't unfortunately - I'm using the full set of commits in testing.

Reply:
I will run the full testsuite when I get a chance - probably this weekend.

Reply:
So, as far as I can tell, the new behaviour (of adding `new` to certain properties) is in fact the correct one. Consider the following simple testcase:

```
public abstract class A {
	public object Value { get; set; }
}

public class A<T> : A {
	public new T Value { get { return default(T); } set {} }
}

public class B : A<string> {
	public new string Value { get { return ""; } set {} }
}
```

According to the C# compiler, both `new` keywords are needed (it gives warnings about hiding things otherwise - even though `B.value` hides an identical `string Value` in the concretized superclass!). Similarly, in the example you posted, I think the property you're looking at actually does hide a property from the base class. I'm happy to be proven wrong on this though - in which case I'll probably have to recheck how I'm computing the signature comparison.

Reply:
Based on the test suite output, I did identify one place where the new code generated an incorrect `new` modifier (on an `explicit` conversion operator); that's been fixed now.

Reply:
Okay, I compiled Visual Studio solutions for the stub code output of two of the major titles in the test suite we're using and didn't get any method hiding errors; in addition as you said Methods.cs now de-compiles correctly (I had completely missed the `new` modifier omission in `TestNewNonVirtualMethod` - good eye). Not exactly a rigorous test but I think we will consider this resolved for now unless I notice anything crop up in my day-to-day usage.